### ISSUE TITLE
tests.tools/run_tests/run_node.sh failure: Call propagation Cancellation With a client stream call, timeout

### ISSUE THREAD

**Issue by dgquintas** [grpc/grpc#5294]
https://grpc-testing.appspot.com/job/gRPC_master/15215/config=tsan,language=node,platform=linux/consoleText

```
...
all(server=0x18ec670, call=0x1ddda08, details=0x1ddda10, initial_metadata=0x1ddda48, cq_bound_to_call=0x192f7d0, cq_for_notification=0x192f7d0, tag=0x1915400)
I0218 00:06:53.437562824    6101 call.c:1450]                grpc_call_start_batch(call=0x7f615805c280, ops=0x1b38100, nops=1, tag=0x18f8750, reserved=(nil))
I0218 00:06:53.437577059    6101 call.c:1165]                ops[0]: RECV_CLOSE_ON_SERVER cancelled=0x18f86f8
I0218 00:06:53.437602494    6101 call.c:1450]                grpc_call_start_batch(call=0x7f615805c280, ops=0x1b38100, nops=1, tag=0x1dd7b30, reserved=(nil))
I0218 00:06:53.437613268    6101 call.c:1165]                ops[0]: RECV_MESSAGE ptr=0x23c9858
I0218 00:06:53.437627854    6101 call_details.c:47]          grpc_call_details_destroy(cd=0x1b38170)
I0218 00:06:53.437637016    6101 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1b381a8)
I0218 00:06:53.437704057    6101 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x18ae588)
I0218 00:06:53.437711544    6109 completion_queue.c:291]     grpc_completion_queue_next(cc=0x192f7d0, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.437754622    6109 completion_queue.c:328]     RETURN_EVENT[0x192f7d0]: OP_COMPLETE: tag:0x18f8750 OK
I0218 00:06:53.437767701    6106 completion_queue.c:291]     grpc_completion_queue_next(cc=0x192f7d0, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.437782945    6106 completion_queue.c:328]     RETURN_EVENT[0x192f7d0]: OP_COMPLETE: tag:0x1dd7b30 OK
       Should correctly cancel a server stream call: 1ms
I0218 00:06:53.438123839    6101 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x2394c18)
I0218 00:06:53.438210049    6107 completion_queue.c:291]     grpc_completion_queue_next(cc=0x192f7d0, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.438411053    6101 metadata_array.c:42]        grpc_metadata_array_init(array=0x7fffa5ef80c0)
I0218 00:06:53.438432866    6101 metadata_array.c:42]        grpc_metadata_array_init(array=0x7fffa5ef80c0)
I0218 00:06:53.438450167    6101 call.c:1450]                grpc_call_start_batch(call=0x7f615805c280, ops=0x1dd7b60, nops=2, tag=0x1910980, reserved=(nil))
I0218 00:06:53.438459782    6101 call.c:1165]                ops[0]: SEND_INITIAL_METADATA
I0218 00:06:53.438467483    6101 call.c:1165]                ops[1]: SEND_STATUS_FROM_SERVER status=3 details=Illegal buffer
I0218 00:06:53.438541778    6108 completion_queue.c:291]     grpc_completion_queue_next(cc=0x192f7d0, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.438560451    6108 completion_queue.c:328]     RETURN_EVENT[0x192f7d0]: OP_COMPLETE: tag:0x1910980 ERROR
I0218 00:06:53.439224108    6101 channel.c:202]              grpc_channel_create_call(channel=0x18f0990, parent_call=(nil), propagation_mask=ffff, cq=0x192f7d0, method=/math.Math/DivMany, host=(null), deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.439355927    6101 metadata_array.c:42]        grpc_metadata_array_init(array=0x7fffa5efb690)
I0218 00:06:53.439372352    6101 metadata_array.c:42]        grpc_metadata_array_init(array=0x191f728)
I0218 00:06:53.439387868    6101 call.c:1450]                grpc_call_start_batch(call=0x18f86f0, ops=0x1916190, nops=2, tag=0x1910980, reserved=(nil))
I0218 00:06:53.439397915    6101 call.c:1165]                ops[0]: SEND_INITIAL_METADATA
I0218 00:06:53.439405196    6101 call.c:1165]                ops[1]: RECV_INITIAL_METADATA ptr=0x191f728
I0218 00:06:53.439499140    6101 metadata_array.c:42]        grpc_metadata_array_init(array=0x18f8148)
I0218 00:06:53.439510410    6109 completion_queue.c:291]     grpc_completion_queue_next(cc=0x192f7d0, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.439517480    6101 call.c:1450]                grpc_call_start_batch(call=0x18f86f0, ops=0x23c9920, nops=1, tag=0x18ae580, reserved=(nil))
I0218 00:06:53.439519570    6107 call.c:1165]                ops[0]: RECV_INITIAL_METADATA ptr=0x7f614c0678e8
I0218 00:06:53.439552897    6101 call.c:1165]                ops[0]: RECV_STATUS_ON_CLIENT metadata=0x18f8148 status=0x18f8160 details=0x18f8168
I0218 00:06:53.439588672    6107 completion_queue.c:328]     RETURN_EVENT[0x192f7d0]: OP_COMPLETE: tag:0x1915400 OK
I0218 00:06:53.439601461    6101 call.c:670]                 grpc_call_cancel(call=0x18f86f0, reserved=(nil))
I0218 00:06:53.439612902    6101 call.c:685]                 grpc_call_cancel_with_status(c=0x18f86f0, status=1, description=Cancelled, reserved=(nil))
I0218 00:06:53.439680425    6109 completion_queue.c:328]     RETURN_EVENT[0x192f7d0]: OP_COMPLETE: tag:0x1910980 OK
I0218 00:06:53.439685507    6106 completion_queue.c:291]     grpc_completion_queue_next(cc=0x192f7d0, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.439726023    6106 completion_queue.c:328]     RETURN_EVENT[0x192f7d0]: OP_COMPLETE: tag:0x18ae580 OK
I0218 00:06:53.439925266    6101 call_details.c:42]          grpc_call_details_init(cd=0x23c9930)
I0218 00:06:53.439938366    6101 metadata_array.c:42]        grpc_metadata_array_init(array=0x23c9968)
I0218 00:06:53.439952520    6101 server.c:1173]              grpc_server_request_call(server=0x18ec670, call=0x23c9928, details=0x23c9930, initial_metadata=0x23c9968, cq_bound_to_call=0x192f7d0, cq_for_notification=0x192f7d0, tag=0x23a1e20)
I0218 00:06:53.440035066    6101 call.c:1450]                grpc_call_start_batch(call=0x7f614c067300, ops=0x19161f0, nops=1, tag=0x23c9830, reserved=(nil))
I0218 00:06:53.440048687    6101 call.c:1165]                ops[0]: RECV_CLOSE_ON_SERVER cancelled=0x23c97d8
I0218 00:06:53.440092251    6101 call_details.c:47]          grpc_call_details_destroy(cd=0x1ddda10)
I0218 00:06:53.440103400    6101 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1ddda48)
I0218 00:06:53.440150459    6108 completion_queue.c:291]     grpc_completion_queue_next(cc=0x192f7d0, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.440158327    6101 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x191f728)
I0218 00:06:53.440166933    6108 completion_queue.c:328]     RETURN_EVENT[0x192f7d0]: OP_COMPLETE: tag:0x23c9830 OK
I0218 00:06:53.440195621    6108 completion_queue.c:291]     grpc_completion_queue_next(cc=0x192f7d0, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
       Should correctly cancel a bidi stream call: 1ms
I0218 00:06:53.440609107    6101 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x18f8148)
I0218 00:06:53.440828652    6101 server.c:1005]              grpc_server_shutdown_and_notify(server=0x18ec670, cq=0x18f0820, tag=(nil))
I0218 00:06:53.440887136    6108 completion_queue.c:328]     RETURN_EVENT[0x192f7d0]: OP_COMPLETE: tag:0x23a1e20 ERROR
I0218 00:06:53.440930574    6101 server.c:1057]              grpc_server_cancel_all_calls(server=0x18ec670)
I0218 00:06:53.440945597    6101 completion_queue.c:379]     grpc_completion_queue_pluck(cc=0x18f0820, tag=(nil), deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 00:06:53.440956720    6101 completion_queue.c:434]     RETURN_EVENT[0x18f0820]: OP_COMPLETE: tag:(nil) OK
I0218 00:06:53.441091853    6101 call_details.c:47]          grpc_call_details_destroy(cd=0x23c9930)
I0218 00:06:53.441104230    6101 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x23c9968)

  Suite duration: 0.012 s, Tests: 4

  193 passing (7s)
  2 pending
  1 failing

  1) Call propagation Cancellation With a client stream call:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
  



Command exited with non-zero status 1
real 8.70
user 2.44
sys 0.26
```

**Comment by dklempner** [comment]
I saw what looked like a similar failure in 15237.

**Comment by murgatroid99** [comment]
OK, it looks like this still isn't fixed. I'm seeing this on my own machine even after quadrupling the timeout.

**Comment by murgatroid99** [comment]
@ctiller can you take a look at this?

The test that is failing here does the following:

 1. Spins up two servers:
    - Server A has a handler that makes no response
    - Server B has a handler that forwards the call to server A, explicitly naming the call it receives as the parent. It then asserts that the call it sends ends with the CANCELLED status (more on that later).
 2. The client makes a call (1) to server B.
 3. When Server B receives it, it makes a call (2) to server A. **It then cancels call 1.**
 4. We assert that call 1 ends and that call 2 ends with CANCELLED status.

To reproduce this failure, I changed line 975 of src/node/test/surface_test.js from `it(` to `it.only(` to only run this test, then just ran the following:

```bash
while ./node_modules/.bin/mocha src/node/test; do echo TEST DONE; done
```

I have generated a complete log of just this test failing. I have separated a few lines of particular note:

```
I0218 16:29:04.650372776   17302 init.c:140]                 grpc_init(void)
I0218 16:29:04.650884579   17302 completion_queue.c:121]     grpc_completion_queue_create(reserved=(nil))
I0218 16:29:04.667517034   17302 server_create.c:46]         grpc_server_create((nil), (nil))
I0218 16:29:04.668171178   17302 server.c:766]               grpc_server_register_completion_queue(server=0x1b4b600, cq=0x1b6bf60, reserved=(nil))
I0218 16:29:04.668187845   17302 completion_queue.c:121]     grpc_completion_queue_create(reserved=(nil))
I0218 16:29:04.668195903   17302 server.c:766]               grpc_server_register_completion_queue(server=0x1b4b600, cq=0x1b6c0f0, reserved=(nil))
I0218 16:29:04.955671932   17302 server_create.c:46]         grpc_server_create((nil), (nil))
I0218 16:29:04.956342201   17302 server.c:766]               grpc_server_register_completion_queue(server=0x1bbf520, cq=0x1b6bf60, reserved=(nil))
I0218 16:29:04.956357112   17302 completion_queue.c:121]     grpc_completion_queue_create(reserved=(nil))
I0218 16:29:04.956365359   17302 server.c:766]               grpc_server_register_completion_queue(server=0x1bbf520, cq=0x1b71af0, reserved=(nil))


  Call propagation
I0218 16:29:04.983573353   17302 server_create.c:46]         grpc_server_create((nil), (nil))
I0218 16:29:04.984211275   17302 server.c:766]               grpc_server_register_completion_queue(server=0x1d60920, cq=0x1b6bf60, reserved=(nil))
I0218 16:29:04.984223669   17302 completion_queue.c:121]     grpc_completion_queue_create(reserved=(nil))
I0218 16:29:04.984231701   17302 server.c:766]               grpc_server_register_completion_queue(server=0x1d60920, cq=0x1c4bf40, reserved=(nil))
I0218 16:29:04.984712550   17302 server_chttp2.c:98]         grpc_server_add_insecure_http2_port(server=0x1d60920, addr=localhost:0)
I0218 16:29:04.985274923   17302 channel_create.c:201]       grpc_insecure_channel_create(target=0x7fffb57546f0, args=0x1d60d30, reserved=(nil))
I0218 16:29:04.985423318   17302 server.c:872]               grpc_server_start(server=0x1d60920)
I0218 16:29:04.985448702   17302 call_details.c:42]          grpc_call_details_init(cd=0x1ba15c0)
I0218 16:29:04.985458393   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x1ba15f8)
I0218 16:29:04.985473805   17302 server.c:1173]              grpc_server_request_call(server=0x1d60920, call=0x1ba15b8, details=0x1ba15c0, initial_metadata=0x1ba15f8, cq_bound_to_call=0x1b6bf60, cq_for_notification=0x1b6bf60, tag=0x1ba1730)
I0218 16:29:04.985747261   17307 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
    Cancellation
I0218 16:29:04.986467129   17302 server_create.c:46]         grpc_server_create((nil), (nil))
I0218 16:29:04.987127473   17302 server.c:766]               grpc_server_register_completion_queue(server=0x1df00b0, cq=0x1b6bf60, reserved=(nil))
I0218 16:29:04.987138822   17302 completion_queue.c:121]     grpc_completion_queue_create(reserved=(nil))
I0218 16:29:04.987146330   17302 server.c:766]               grpc_server_register_completion_queue(server=0x1df00b0, cq=0x1defd60, reserved=(nil))
I0218 16:29:04.988247541   17302 server_chttp2.c:98]         grpc_server_add_insecure_http2_port(server=0x1df00b0, addr=localhost:0)
I0218 16:29:04.988351101   17302 server.c:872]               grpc_server_start(server=0x1df00b0)
I0218 16:29:04.988380003   17302 call_details.c:42]          grpc_call_details_init(cd=0x1ba1770)
I0218 16:29:04.988389125   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x1ba17a8)
I0218 16:29:04.988400484   17302 server.c:1173]              grpc_server_request_call(server=0x1df00b0, call=0x1ba1768, details=0x1ba1770, initial_metadata=0x1ba17a8, cq_bound_to_call=0x1b6bf60, cq_for_notification=0x1b6bf60, tag=0x1df0010)
I0218 16:29:04.988456432   17308 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:04.988485648   17302 channel_create.c:201]       grpc_insecure_channel_create(target=0x7fffb5754580, args=0x1ba1d50, reserved=(nil))
I0218 16:29:04.988711472   17302 channel.c:202]

#######
# CALL 1
#######
grpc_channel_create_call(channel=0x1bc9a10, parent_call=(nil), propagation_mask=ffff, cq=0x1b6bf60, method=/TestService/ClientStream, host=(null), deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
#######

I0218 16:29:04.989045900   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x7fffb5754900)
I0218 16:29:04.989061678   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x1bb5f68)
I0218 16:29:04.989074954   17302 call.c:1450]                grpc_call_start_batch(call=0x1bb5650, ops=0x1bb5e50, nops=2, tag=0x1bb5fb0, reserved=(nil))
I0218 16:29:04.989089883   17302 call.c:1165]                ops[0]: SEND_INITIAL_METADATA
I0218 16:29:04.989097100   17302 call.c:1165]                ops[1]: RECV_INITIAL_METADATA ptr=0x1bb5f68
I0218 16:29:04.989159008   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x1bb60c8)
I0218 16:29:04.989173382   17302 call.c:1450]                grpc_call_start_batch(call=0x1bb5650, ops=0x1bb5e50, nops=2, tag=0x1bb6050, reserved=(nil))
I0218 16:29:04.989181621   17302 call.c:1165]                ops[0]: RECV_MESSAGE ptr=0x1ba1888
I0218 16:29:04.989189112   17302 call.c:1165]                ops[1]: RECV_STATUS_ON_CLIENT metadata=0x1bb60c8 status=0x1bb60e0 details=0x1bb60e8
I0218 16:29:04.989840027   17308 call.c:1165]                ops[0]: RECV_INITIAL_METADATA ptr=0x7f6830007d98
I0218 16:29:04.989983749   17308 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1df0010 OK
I0218 16:29:04.990106623   17309 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:04.990885361   17302 call_details.c:42]          grpc_call_details_init(cd=0x1bca000)
I0218 16:29:04.990901209   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x1bca038)
I0218 16:29:04.990912710   17302 server.c:1173]              grpc_server_request_call(server=0x1df00b0, call=0x1bc9ff8, details=0x1bca000, initial_metadata=0x1bca038, cq_bound_to_call=0x1b6bf60, cq_for_notification=0x1b6bf60, tag=0x1d60cc0)
I0218 16:29:04.991205009   17302 call.c:1450]                grpc_call_start_batch(call=0x7f68300077b0, ops=0x1df0040, nops=1, tag=0x1bb6120, reserved=(nil))
I0218 16:29:04.991220723   17302 call.c:1165]                ops[0]: RECV_CLOSE_ON_SERVER cancelled=0x1bb6008
I0218 16:29:04.991639695   17302 channel.c:202]              

#######
# CALL 2
#######
grpc_channel_create_call(channel=0x1ba9de0, parent_call=0x7f68300077b0, propagation_mask=ffff, cq=0x1b6bf60, method=/TestService/ClientStream, host=(null), deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
#######

I0218 16:29:04.991798022   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x7fffb5751440)
I0218 16:29:04.991812234   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x1de0758)
I0218 16:29:04.991822879   17302 call.c:1450]                grpc_call_start_batch(call=0x1ddfe20, ops=0x1de0620, nops=2, tag=0x1de0780, reserved=(nil))
I0218 16:29:04.991831268   17302 call.c:1165]                ops[0]: SEND_INITIAL_METADATA
I0218 16:29:04.991838322   17302 call.c:1165]                ops[1]: RECV_INITIAL_METADATA ptr=0x1de0758
I0218 16:29:04.991894504   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x1de09f8)
I0218 16:29:04.991909133   17302 call.c:1450]                grpc_call_start_batch(call=0x1ddfe20, ops=0x1de0620, nops=2, tag=0x1de0870, reserved=(nil))
I0218 16:29:04.991917368   17302 call.c:1165]                ops[0]: RECV_MESSAGE ptr=0x1de0f28
I0218 16:29:04.991925386   17302 call.c:1165]                ops[1]: RECV_STATUS_ON_CLIENT metadata=0x1de09f8 status=0x1de0a10 details=0x1de0a18

#######
# CALL 1 CANCELLED
#######
I0218 16:29:04.991971841   17302 call.c:670]                 grpc_call_cancel(call=0x1bb5650, reserved=(nil))
I0218 16:29:04.991984763   17302 call.c:685]                 grpc_call_cancel_with_status(c=0x1bb5650, status=1, description=Cancelled, reserved=(nil))
#######

I0218 16:29:04.992042836   17302 call_details.c:47]          grpc_call_details_destroy(cd=0x1ba1770)
I0218 16:29:04.992054227   17302 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1ba17a8)

#######
# CALL 2 CANCELLED
#######
I0218 16:29:04.992091611   17307 call.c:670]                 grpc_call_cancel(call=0x1ddfe20, reserved=(nil))
I0218 16:29:04.992108653   17307 call.c:685]                 grpc_call_cancel_with_status(c=0x1ddfe20, status=1, description=Cancelled, reserved=(nil))
#######

#######
# CALL 1 RECV_INITIAL_METADATA, RECV_STATUS, RECV_MESSAGE
#######
I0218 16:29:04.992171139   17307 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1bb5fb0 OK
I0218 16:29:04.992182602   17309 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1bb6050 OK
#######

I0218 16:29:04.992203313   17310 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:04.992228691   17310 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1bb6120 OK
I0218 16:29:04.992308492   17302 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1bb5f68)
I0218 16:29:04.992336516   17308 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:04.992385074   17308 call.c:1165]                ops[0]: RECV_INITIAL_METADATA ptr=0x7f683000aef8
I0218 16:29:04.992450190   17308 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1ba1730 OK
Parent call finished
I0218 16:29:04.992607164   17302 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1bb60c8)
I0218 16:29:04.992631003   17307 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:04.993770770   17309 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:04.993904332   17302 call_details.c:42]          grpc_call_details_init(cd=0x1ba1830)
I0218 16:29:04.993921430   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x1ba1868)
I0218 16:29:04.993933396   17302 server.c:1173]              grpc_server_request_call(server=0x1d60920, call=0x1ba1828, details=0x1ba1830, initial_metadata=0x1ba1868, cq_bound_to_call=0x1b6bf60, cq_for_notification=0x1b6bf60, tag=0x1df0010)
I0218 16:29:04.994090455   17302 call.c:1450]                grpc_call_start_batch(call=0x7f683000a910, ops=0x1bc96e0, nops=1, tag=0x1bb6080, reserved=(nil))
I0218 16:29:04.994105496   17302 call.c:1165]                ops[0]: RECV_CLOSE_ON_SERVER cancelled=0x1bc99f8
I0218 16:29:04.994141766   17302 call_details.c:47]          grpc_call_details_destroy(cd=0x1ba15c0)
I0218 16:29:04.994152552   17302 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1ba15f8)
#######
# TEST FAILURE (and 2 seconds of until the next log line)
#######
      1) With a client stream call
I0218 16:29:06.996689812   17302 server.c:1005]              

#######
# SERVER B SHUTDOWN AND CANCEL ALL CALLS
#######
grpc_server_shutdown_and_notify(server=0x1df00b0, cq=0x1defd60, tag=(nil))
I0218 16:29:06.996759013   17307 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1d60cc0 ERROR
D0218 16:29:06.996819578   17309 chttp2_transport.c:683]     got goaway [0]: 53 65 72 76 65 72 20 73 68 75 74 64 6f 77 6e 'Server shutdown'
I0218 16:29:06.996826121   17302 server.c:1057]              grpc_server_cancel_all_calls(server=0x1df00b0)
#######

I0218 16:29:06.996859900   17302 completion_queue.c:379]     grpc_completion_queue_pluck(cc=0x1defd60, tag=(nil), deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:06.996871127   17302 completion_queue.c:434]     RETURN_EVENT[0x1defd60]: OP_COMPLETE: tag:(nil) OK
I0218 16:29:06.996926266   17310 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:06.996958891   17302 call_details.c:47]          grpc_call_details_destroy(cd=0x1bca000)
I0218 16:29:06.996972106   17302 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1bca038)
I0218 16:29:06.997477712   17302 server.c:1005]              

#######
# SERVER A SHUTDOWN AND CANCEL ALL CALLS
#######
grpc_server_shutdown_and_notify(server=0x1d60920, cq=0x1c4bf40, tag=(nil))
I0218 16:29:06.997504892   17310 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1df0010 ERROR
I0218 16:29:06.997544660   17302 server.c:1057]              grpc_server_cancel_all_calls(server=0x1d60920)
#######

D0218 16:29:06.997554341   17309 chttp2_transport.c:683]     got goaway [0]: 53 65 72 76 65 72 20 73 68 75 74 64 6f 77 6e 'Server shutdown'
I0218 16:29:06.997588303   17302 completion_queue.c:379]     grpc_completion_queue_pluck(cc=0x1c4bf40, tag=(nil), deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))

#######
# CALL 2 RECV_INITIAL_METADATA, RECV_STATUS, RECV_MESSAGE
I0218 16:29:06.997595225   17309 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1bb6080 OK
I0218 16:29:06.997606557   17302 completion_queue.c:434]     RETURN_EVENT[0x1c4bf40]: OP_COMPLETE: tag:(nil) OK
#######

I0218 16:29:06.997775795   17308 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:06.997797756   17302 call_details.c:47]          grpc_call_details_destroy(cd=0x1ba1830)
I0218 16:29:06.997812957   17302 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1ba1868)
I0218 16:29:06.997831327   17307 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:06.997847085   17307 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1de0780 OK
I0218 16:29:06.997848887   17308 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1de0870 OK
I0218 16:29:06.998221711   17302 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1de0758)
Child call cancelled correctly
I0218 16:29:06.998755812   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x7fffb5751480)
I0218 16:29:06.998776804   17302 metadata_array.c:42]        grpc_metadata_array_init(array=0x7fffb5751470)
I0218 16:29:06.998791788   17302 call.c:1450]                grpc_call_start_batch(call=0x7f68300077b0, ops=0x1bc9f30, nops=2, tag=0x1bc9c40, reserved=(nil))
I0218 16:29:06.998801558   17302 call.c:1165]                ops[0]: SEND_INITIAL_METADATA
I0218 16:29:06.998808778   17302 call.c:1165]                ops[1]: SEND_STATUS_FROM_SERVER status=1 details=Cancelled
I0218 16:29:06.998833432   17310 completion_queue.c:291]     grpc_completion_queue_next(cc=0x1b6bf60, deadline=gpr_timespec { tv_sec: 9223372036854775807, tv_nsec: 0, clock_type: 1 }, reserved=(nil))
I0218 16:29:06.998844871   17310 completion_queue.c:328]     RETURN_EVENT[0x1b6bf60]: OP_COMPLETE: tag:0x1bc9c40 ERROR
I0218 16:29:06.998923064   17302 metadata_array.c:47]        grpc_metadata_array_destroy(array=0x1de09f8)
```

This boils down to the following sequence of events:

 1. Call 1 is created.
 2. Call 2 is created.
 3. Call 1 is cancelled.
 4. Call 2 is cancelled.
 5. Call 1 receives its cancellation status.
 6. **Nothing relevant happens for 2 seconds.**
 7. The test fails
 8. Both servers shut down and cancel all calls.
 9. Call 2 finally receives its cancellation status.

**Comment by murgatroid99** [comment]
Note: I have only seen this error with client streaming calls. That mainly implies that the client waits to receive a message but does not send a message or half-close.

**Comment by jtattermusch** [comment]
Seen again here:
https://grpc-testing.appspot.com/job/gRPC_master/15327/config=dbg,language=node,platform=linux/testReport/junit/(root)/tests/tools_run_tests_run_node_sh/

**Comment by murgatroid99** [comment]
Seen again in 3/6 test runs at https://grpc-testing.appspot.com/job/gRPC_master/15408

**Comment by murgatroid99** [comment]
This was fixed by #5625.
